File: linguagem_java/encapsulamento/empresa/Empresa.java
package encapsulamento.empresa;

public class Empresa {
    // Atributos da classe Empresa
    private String nomeDaEmpresa;

    // Construtor da classe Empresa
    public Empresa(String nomeDaEmpresa) {
        setNomeDaEmpresa(nomeDaEmpresa);
    }

    // Getter e setter de nomeDaEmpresa
    public String getNomeDaEmpresa() {
        return nomeDaEmpresa;
    }

    public void setNomeDaEmpresa(String nomeDaEmpresa) {
        this.nomeDaEmpresa = nomeDaEmpresa;
    }

}

File: linguagem_dart/encapsulamento/empresa/empresa.dart
class Empresa {
  // Atributos da classe Empresa
  String _nomeDaEmpresa = "";

  // Construtor da classe Empresa
  Empresa(String nome) {
    setNomeDaEmpresa = nome;
  }

  // Getter e setter de NomeDaEmpresa
  String get getNomeDaEmpresa => _nomeDaEmpresa;

  set setNomeDaEmpresa(String nomeDaEmpresa) {
    _nomeDaEmpresa = nomeDaEmpresa;
  }
}
